src/sentry/grouping/strategies/newstyle.py

```diff
@@ -733,6 +740,83 @@ def chained_exception(
     return rv
 
 
+# See https://github.com/getsentry/rfcs/blob/main/text/0079-exception-groups.md#sentry-issue-grouping
+def filter_exceptions_for_exception_groups(
+    exceptions: List[SingleException], context: GroupingContext, event: Event, **meta: Any
+) -> List[SingleException]:
+
+    # This function only filters exceptions if there are at least two exceptions.
+    if len(exceptions) <= 1:
+        return exceptions
+
+    # Reconstruct the tree of exceptions if the required data is present.
+    nodes = {}
+    for exception in reversed(exceptions):
+        exception._children = []
+        mechanism: Mechanism = exception.mechanism
+        if mechanism and mechanism.exception_id is not None:
+            nodes[mechanism.exception_id] = exception
+            if mechanism.parent_id is not None:
+                nodes[mechanism.parent_id]._children.append(exception)
+
+    # If the nodes list is empty, then we don't have the required data to reconstruct the tree.
+    if len(nodes) == 0:
+        return exceptions
+
+    # Traverse the tree recursively from the root node to get all "top-level exceptions"
```

Comment:
Yeah, they're not exactly "leaves", because they can have children.  They're the first exceptions (from the root of the tree down) that are not exception groups themselves.

I can add some docs with examples, as you suggested.